sockets/webget.c

```diff
@@ -0,0 +1,133 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <inttypes.h>
+#include <string.h>
+
+int main(int argc, char ** argv)
+{
+	//port used to connect to socket
+	char* server_port = "80";
+	//ip address of server http request will be sent to
+	char* server_ip = "173.194.121.36";
+	char * host_name = "http://www.google.com";
+	//socket int used to connect
+	int sockfd, rc;
+	//used to store formatted http request
+	char * http_format_req; 
+	struct addrinfo hints, *server;
+	char *message = "";
+	//bytes received from server
+	int bytes_received = -1;
+	//data received from http request--will be printed out to user
+	char * recv_data = (char *)malloc(sizeof(char*)*1000);
+	int o;
+
+        /* Command line args:
+                -p port
+                -h host name or IP
+		-i http request
```

Comment:
Why do you need both hostname and server IP?  You should be able to figure out one from the other after calling getaddrinfo().  (also this explanation of -i doesn't make sense to me)